Source/Core/Core/Core.cpp

```diff
@@ -853,10 +854,9 @@ void RunOnCPUThread(Core::System& system, Common::MoveOnlyFunction<void()> funct
   else if (wait_for_completion)
   {
     // Queue the job function followed by triggering the event.
-    s_cpu_thread_job_finished.Reset();
-    system.GetCPU().AddCPUThreadJob([&function] {
+    system.GetCPU().AddCPUThreadJob([&function, &cpu_thread_job_finished] {
       function();
-      s_cpu_thread_job_finished.Set();
+      cpu_thread_job_finished.Set();
```

Comment:
I think this enables a race where the destruction of the `Event` can happen in the middle of `Set`.
(Same thing that #14079 recently fixed)

Reply:
You could use `packaged_task` like AsyncWorkThread's `PushBlocking` does.

Or if `Event::Set` was changed to hold the mutex for the *entire* call I think the "Set() for destruction" pattern would become race free. https://github.com/dolphin-emu/dolphin/blob/56532c850f0f00a129207fae1d2ef6867b51d9cc/Source/Core/Common/Event.h#L38